ArtOfIllusion/src/artofillusion/math/TriangleMath.java

```diff
@@ -0,0 +1,134 @@
+/* Copyright (C) 2016 by Petri Ihalainen
+
+   This program is free software; you can redistribute it and/or modify it under the
+   terms of the GNU General Public License as published by the Free Software
+   Foundation; either version 2 of the License, or (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful, but WITHOUT ANY
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+   PARTICULAR PURPOSE.  See the GNU General Public License for more details. */
+
+package artofillusion.math;
+
+import artofillusion.*;
+import artofillusion.ui.*;
+import artofillusion.math.*;
+
+/**
+ *  TriangleProps contains frequently used calculations about triangles
+ */
+public class TriangleMath
+{
+	/** Constructor for TriangleProps */
+	public void TriangleMath(){}
+	
+	//********************************//
+	//*   METHODS FOR 3D TRIANGLES   *//
+	//********************************//
+
+	/** Cross product of two side vectors in 3D*/ 
+	public Vec3 cross(Vec3 A, Vec3 B, Vec3 C)
```

Comment:
Following on from the bit about a private constructor, you would then mark all of the methods as static, like this:

public static Vec3 cross(Vec3 A, Vec3 B, Vec3 C)
{
    return B.minus(A).cross(C.minus(A));
}
It might also be worth considering potential method name collisions with, eg. Vec3.